### PR TITLE
Js nonlinear conductivity 2

### PR DESCRIPTION
**Pull request by jstebel**
- Pass mechanics fields to flow.
- Add some new mechanics fields need for computation of conductivity.
- Add initial stress field (by @Paulie14)
- Add assembly classes to HM_Iterative, remove calling of Field::value().

### CODE REVIEW

File: src/coupling/assembly_hm.hh [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+        fe_values_.reinit(cell.elm());
+
+        // compute pressure error
+        unsigned int k=0;
+        for (auto p : this->bulk_points(element_patch_idx) )
+        {
+            double new_p = eq_data_->flow_->eq_fields().field_ele_pressure(p);
+            double old_p = eq_fields_->old_iter_pressure(p);
+            eq_data_->p_dif2 += pow(new_p - old_p, 2)*fe_values_.JxW(k);
```
Comment (jbrezmorf) [inline src/coupling/assembly_hm.hh:165]: Rather use multiplication for small integer powers, 10x times faster:
https://stackoverflow.com/questions/2940367/what-is-more-efficient-using-pow-to-square-or-just-multiply-it-with-itself.

File: tests/50_mechanics/31_3d_block_initial_stress.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Test goal: 3D linear elastic simple loading problem.
```
Comment (jbrezmorf) [inline tests/50_mechanics/31_3d_block_initial_stress.yaml:1]: Add to config.yaml.

**Review by jbrezmorf — COMMENTED** [review]
Timelimit problem in debug build is related to the changes in general assembly routines.

File: src/mechanics/elasticity.cc [hunk 3]
```diff
@@ -407,6 +424,8 @@ Elasticity::~Elasticity()
 
 void Elasticity::update_output_fields()
 {
+    eq_fields_->set_time(time_->step(), LimitSide::right);
+    
     // update ghost values of solution vector
 	eq_fields_->output_field_ptr->vec().local_to_ghost_begin();
```
[the remark this answers is not in the document]
Comment (jstebel) [reply src/mechanics/elasticity.cc:430]: Fixed

**Pull request merged**